torch/csrc/distributed/c10d/Types.hpp

```diff
@@ -67,4 +69,13 @@ struct BarrierOptions {
   std::chrono::milliseconds timeout = kUnsetTimeout;
 };
 
+struct DistributedBackendOptions {
+  c10::intrusive_ptr<::c10d::Store> store;
+  int group_rank;
+  int group_size;
+  std::chrono::duration<float> timeout;
+  int group_id;
```

Comment:
Group ID should be a string and use `group_name` from the python side. We don't want to leak the details that this is a counter as it's going to change in the near future to a random string.

Reply:
Done